### PR TITLE
perf: use indices for _some { equal: ... } filters

### PR DESCRIPTION
**Pull request by Yogu**
We now generate a simple  expression, which can be executed using indices by ArangoDB.

Previously, we had a conditional around object.list that would convert NULL values to []. This prevented the index from being used. The IN operator is always false for non-array values, so we do not need this conditional.

Also simplifies the _every and _none quantifies, but they currently cannot use indices.

### CODE REVIEW

File: src/database/arangodb/aql-generator.ts [hunk 1]
```diff
@@ -1319,39 +1319,49 @@ function getQuantifierFilterUsingArrayComparisonOperator(
 
     const valueFrag = processNode(conditionNode.rhs, context);
 
-    let fieldValueFrag: AQLFragment;
+    let fieldValueFrag: () => AQLFragment;
```
Comment (Etienne-Buschong) [inline src/database/arangodb/aql-generator.ts:1322]: Why did you change this field to a function that resolves to an `AQLFragment` ?

**Review by Etienne-Buschong — APPROVED** [review]

File: src/database/arangodb/aql-generator.ts (hunk 1 above)
Comment (Yogu) [reply src/database/arangodb/aql-generator.ts]: the value of `isSafeList` can now change *after* the conditional here. I could untangle the logic so it first fully calculates `isSafeList`, then calculates the `fieldValueFrag`, and then uses it. However, the logic of whether or not `isSafeList` can safely be set to `false` depends on how the `fieldValueFrag` is used, so I think it's clearer to set that flag to false where we use the `fieldValueFlag`.

I just changed it a bit again, renamed it to `fieldValueFragFn`, added a parameter `useSafeListWrapper` to make it more explicit, and added comments

Do you think it's easier to understand now?

**Review by Etienne-Buschong — APPROVED** [review]
